### ISSUE TITLE
time display option

### ISSUE THREAD

**Issue by Tuetenk0pp** [vabene1111/recipes#382]
**Is your feature request related to a problem? Please describe.**
Sometimes you have a recipe with ridiculous amounts of time going into it, e.g. a Baguette recipe with 72h proofing time.

**Describe the solution you'd like**
I'd like to have an option to turn on time conversion so that whenever I give a time in minutes, it would display a readable equivalent:
45 min -> 45 min
90 min -> 90 min
150 min -> 2 h 30 min
120 min -> 2 h
4320 min -> 72 h

**Describe alternatives you've considered**
Fractions:
150 min -> 1 1/2 h

**Additional context**
Wether you would like 90 min to display as 90 min, 1h 30 min or as 1,5h is just personal preference. Or is there a standard somewhere that this could follow?

**Comment by vabene1111** [comment]
yes this is definitely something that needs to be added to make longer waiting times easier to read. Its probably also very easy to implement but i haven't gotten around to it yet. I think the standard would be that everything over 60 minutes is converted to hours and minutes.

Thanks for reporting.

**Comment by hotdog218** [comment]
I also would love to see this implemented! Would be nice for bread recipes that have lots of proofing stages.

**Comment by wisdomkhan** [comment]
I would like to get started on this one please. Guide me.

**Comment by vabene1111** [comment]
You will need to find the appropriate code location (likely recipe view. Vue) and write a utility function to convert the displayed time to a more convenient to read format.to test your changes look at the link I send your previously.

**Comment by wisdomkhan** [comment]
https://github.com/vabene1111/recipes/blob/493d553050e9ebd581f6c4a67b70d860b50a34c2/cookbook/views/views.py#L154

This one?

**Comment by vabene1111** [comment]
No you want to work on a frontend component, those are either HTML or vue files. The particular one you are looking for is  Recipe View.vue

If you are not sure how to implement this into the application you could just write the Javascript function and PR it to utils.js and I can implement it into the frontend later

**Comment by wisdomkhan** [comment]
`function timeConvert(n){
    var num = n;
    if (num <= 90) {
        return num + " min";
    }
    else {
        var hours = num / 60;
        var rhours = Math.floor(hours);
        var mins = (hours - rhours) * 60;
        var rmins = Math.round(mins);
        if (mins == 0) {
            return rhours + " h";
        }
        else {
            return rhours + " h " + rmins + " min";
        }
    }
}`

Something like this?

**Comment by vabene1111** [comment]
@wisdomkhan i appreciate you trying to help in getting this project forward and i am willing to invest some time to help new contributers forward but sadly i have very limited time right now and working like this (and the last PR) takes substantially more time for me than simply fixing the issue myself. I am sorry, i would love to have the time to properly go trough this with you but i simply cant right now. 

As a general advice for your contributions try to understand the issue from what is already written there and if you have questions formulate those precisely. If someone leaves an easy issue open its likely due to a lack of time or priority and most maintainers will not have the time to guide you trough this step by step. If you decide to implement something on an issue like this one and dont understand how to best do it i peronally think the best way is to document some clear results. In a case like this particular one that would mean post the code commented, properly formatted as a code block

```javascript
function timeConvert(n){ 
var num = n;
 if (num <= 90) { 
return num + " min";
 } else {
 var hours = num / 60; 
var rhours = Math.floor(hours); 
var mins = (hours - rhours) * 60; 
var rmins = Math.round(mins); 
if (mins == 0) { 
return rhours + " h"; 
} else { 
return rhours + " h " + rmins + " min";
 } 
} 
}
```
Then it is important clearly communicate what your changes do. If its a UI change make sure to include a screenshot, if its code like this write all relevant input cases for the function (e.g. entering 0 minutes, 1 minute, 30 minutes, 70 minutes, and likely something with more than 24 hours) and show that it does for each case what it is supposed to do. This way the maintainer does not need to go into a javascript console themself and test out all those cases and can quickly see that the suggested change is working as intended. Optionally many maintainers expect that you quickly elaborate on considered alternatives and a small explanaition on why you decided to implement something the way you did (probably not on something as simple as this but if you start doing bigger contributions)

I will try to update the contribution guide so its easier to understand where you can get started and where you need to look depending on what you want to edit. Again, i am sorry for not having the time to help you further on this particular contribution, from a first look your code looks like it does what it is supposed to do so i might be able to use it when fixing this issue altough there might be some need for localization. Hope this post helps to guide your future contributions, please dont feel demotivated by this but on small projects like this with only a single maintainer working on merging PRs there is often simply not enough time to properly help beginners getting started.

**Comment by wisdomkhan** [comment]
I understand it perfectly sir. No need to be sorry. I'd keep your guidelines in mind for future contributions. Thank you so much for your guidance and time. All the best! 😄

**Comment by dr-froehlich** [comment]
**Is your feature request related to a problem? Please describe.**
[Sometimes you have a recipe with ridiculous amounts of time going into it, e.g. a Baguette recipe with 72h proofing time.] For bread baking e.g. on Saturday preparation steps may have to be performed on Thursday and Friday, adding the days would add clarity.

**Describe the solution you'd like**
In addition to adjusting the time display format it would be helpful to also display the day of the week for each of the involved steps. Example: 17:45 Thursday instead of just 17:45. 

**Describe alternatives you've considered**
Ideally days are displayed as soon as beginning of step 1 and ending of the last step are not on the same day.

Just learned of this very cool project, thanks for the great work!

**Comment by vabene1111** [comment]
yes this makes sense. I explicitly added the time feature for bread baking and actually showing the day not just the time really makes sense

**Comment by mkieledunsche** [comment]
Just looked into the code (Tandor2.0.0-alpha-5) and saw:
[✓] `function calculateHourMinuteSplit(amount)` is implemented in [utils.js](https://github.com/TandoorRecipes/recipes/blob/2.0.0-alpha-5/vue/src/utils/utils.js)
[✓] import of calculateHourMinuteSplit done in [RecipeViewComponent.vue#L170][(https://github.com/TandoorRecipes/recipes/blob/2.0.0-alpha-5/vue/src/components/RecipeViewComponent.vue](https://github.com/TandoorRecipes/recipes/blob/2.0.0-alpha-5/vue/src/components/RecipeViewComponent.vue#L170))
[✓] workingtime and waiting time are computed using calculateHourMinuteSplit() in [RecipeViewComponent.vue#L170][(https://github.com/TandoorRecipes/recipes/blob/2.0.0-alpha-5/vue/src/components/RecipeViewComponent.vue](https://github.com/TandoorRecipes/recipes/blob/2.0.0-alpha-5/vue/src/components/RecipeViewComponent.vue#L170)

Recipe view though does not show the hour-minute-splite in the recipe view (screenshot below)
Let me know if there is anyting I can do to get this topic moved towards closure.

![Image](https://github.com/user-attachments/assets/0f604a91-7486-4676-bdd4-7db398cc6858)

**Comment by mkieledunsche** [comment]
Regarding the topic mentioned by @dr-froehlich would suggest to change [timer.vue](https://github.com/TandoorRecipes/recipes/blob/2.0.0-alpha-5/vue3/src/components/display/Timer.vue)

Would suggest to add a method to the Timer.vue or in case there's a possiblity seen for reuse to add as export funcition to utils.js

```
methods: {
	formatFinishTime(durationSeconds) {
		const now = DateTime.now();
		const target = now.plus({ seconds: durationSeconds });

		let timeString = target.toLocaleString(DateTime.TIME_SIMPLE);

		const daysDifference = Math.floor(
			target.startOf('day').diff(now.startOf('day'), 'days').days
		);
		
		if (daysDifference >= 1) {
			timeString += ` +${daysDifference}d`;
		}

	return timeString;
	}
}
```

replace line 5 `{{$t('FinishedAt')}} {{ DateTime.now().plus({'seconds': durationSeconds}).toLocaleString(DateTime.TIME_SIMPLE) }}`
with `{{$t('FinishedAt')}} {{ formatFinishTime(durationSeconds) }}`

Done some testing for debugging and looks fine:
![Image](https://github.com/user-attachments/assets/c2ec023e-01c2-47da-919a-198109e53f6c)

@vabene1111 let me know if you're ok for me to try go for changing the code of the timer.vue or add it to utils.js

**Comment by vabene1111** [comment]
looks good to me, feel free to implement something like that. Please try to keep it as simple as possible and make sure its localized. 

Also do not use the `vue` folder its for tandoor 1 and will be deleted, all new stuff is in the `vue3` folder. 

Thanks for looking into this.

**Comment by dr-froehlich** [comment]
Opened #4785 for this.

Recipe and step times render as `1 h 30 min` above the hour and `72 h` for a long proof
instead of `4320 min` — at both recipe detail layouts, the recipe card chip (which
previously showed a bare number with no unit at all) and the step timer button. No day
rollover, matching the table earlier in this thread. It sits behind a per-user preference
that defaults off, so nothing changes for existing users unless they turn it on.

One thing worth noting for anyone arriving here from a search: #3751 and #3778 are linked
above and look like they closed this, but what they add is the localized `+2 Days` suffix on
the running step timer's "Finished at" line. That is a different surface, and it works — the
recipe page itself still shows raw minutes today.

Happy to reshape the PR: the test tooling is a separate commit that can be dropped, and the
default is a one-line change if on-by-default would be preferred.

**Comment by dr-froehlich** [comment]
The other half of my original ask here — the weekday on each step — is not in #4785 by design, since it needs an anchor to a wall clock that the recipe page doesn't have today. I've proposed it separately on #1882 (Time Rework), where it fits the "multi day recipes" item: https://github.com/TandoorRecipes/recipes/issues/1882#issuecomment-5276392328

Briefly: you give the recipe a finish time and it back-chains a start time per step, so a bake due Sunday morning tells you when to mix the levain on Thursday. It ended up using both alternatives I floated in 2022 — weekday names near in, and the explicit `(−9 d)` offset once the span passes a week and a bare weekday stops being unambiguous. Frontend only, and it picks the finish time up from the meal planner automatically.

**Comment by dr-froehlich** [comment]
Small update on this: #4785 was closed in favour of a simpler implementation without the user
preference, which is now open as #4810 — recipe and step times rendered as hours
and minutes unconditionally, matching the table in the original request here.
